src/radio/utils/PlaylistParser.js

```diff
@@ -0,0 +1,52 @@
+/* @flow */
+
+const https = require('https');
+const querystring = require('querystring');
+
+export default function parsePlaylist(
+  id: string,
+  token: string = '',
+  resultArray: Array<string> = []
+): Promise<Array<string>> {
+  const key = 'AIzaSyCSPw6aZyX0sdH39Lm_Bzf2SRMdBcqNcp4';
+  let getUrl = 'https://www.googleapis.com/youtube/v3/playlistItems?';
+  const youtubeUrl = 'https://www.youtube.com/watch?v=';
+  const options: Object = {
+    playlistId: id,
+    maxResults: 50,
+    part: 'snippet',
+    key: key,
+  };
+  if (token) {
+    options.pageToken = token;
+  }
+  getUrl += querystring.stringify(options);
+  return new Promise(resolve => {
+    https.get(getUrl, res => {
+      const buffer = [];
+      res.on('data', chunk => {
+        buffer.push(chunk);
+      });
+      res.on('end', () => {
+        const data = Buffer.concat(buffer).toString();
+        const parsedData = JSON.parse(data);
+        for (const item of parsedData.items) {
```

Comment:
we could maybe make this functional

```js
resultArray.push(...parsedData.items
  .filter(item => item.snippet.resourceId.kind === 'youtube#video')
  .filter(item => !['Private video', 'Deleted video'].includes(item.snippet.title))
  .map(item => BASE_YOUTUBE_URL + item.snippet.resourceId.videoId)
);
```
(however prettier formats this)
although it may be less readable
another approach is just the check-error-then-bail approach
like `if (/* not a youtube video */) continue;`